torch/csrc/jit/passes/quantization.cpp

```diff
@@ -202,35 +202,61 @@ graph(%input, %weight, %bias, %4):
       matmul_add};
 };
 
-bool isBiasOfConvOrLinear(Value* v) {
+// Check if `use` is an aten function of name `func_name` and if value
+// `v` is the nth argument of the function
+bool isAtenFuncNthArg(
+    Value* v,
+    Node* use,
+    const std::string& func_name,
+    int n) {
+  return use->kind() == Symbol::aten(func_name) && v == use->inputs().at(n);
+}
+
+// Check if `use` is a CallFunction of name `func_name` and if value
+// `v` is the nth argument of the function
+bool isCallFunctionNthArg(
+    Value* v,
+    Node* use,
+    const std::string& func_name,
+    int n) {
+  return use->kind() == prim::CallFunction &&
+      getFuncName(use->inputs()[0]) == func_name && v == use->inputs().at(n);
+}
+
+struct FuncArg {
+  std::string func_name;
+  int arg_index;
+};
+
+// Check any use of `v` matches the aten function call
+// or CallFunction patterns
+bool matchArgPattern(
+    Value* v,
+    const std::vector<FuncArg>& aten_func_args,
+    const std::vector<FuncArg>& call_func_args) {
   for (const Use& u : v->uses()) {
-    if (u.user->kind() == Symbol::aten("conv2d")) {
-      if (v == u.user->inputs().at(2)) {
+    for (const auto& func_arg : aten_func_args) {
+      if (isAtenFuncNthArg(v, u.user, func_arg.func_name, func_arg.arg_index)) {
         return true;
       }
-    } else if (u.user->kind() == prim::CallFunction) {
-      auto func_name = getFuncName(u.user->inputs()[0]);
-      if (func_name == "linear" && v == u.user->inputs().at(3)) {
+    }
+
+    for (const auto& func_arg : call_func_args) {
+      if (isCallFunctionNthArg(
+              v, u.user, func_arg.func_name, func_arg.arg_index)) {
         return true;
       }
     }
   }
   return false;
 }
 
+bool isBiasOfConvOrLinear(Value* v) {
+  return matchArgPattern(v, {{"conv2d", 2}, {"linear", 2}}, {{"linear", 3}});
```

Comment:
This is concerning, because for `{"linear", 2}` we would say that it is both bias and weight. That's probably incorrect. I think it might work correctly now as a side effect of how we invoke these two functions, but this should not be that fragile.

Reply:
first vector is for aten function call, second is for `CallFunction` call, so although {"linear", 2} appears twice, it mean different things

Reply:
@jerryzh168 Can we make this more explicit at the callsites? The separation with the braces is pretty subtle here and I'm not surprised it's confusing.

Ideally we would have something like python kwargs. I think we can emulate this in C++ with something like:

```
using AtenFuncArgs = std::vector<FuncArg>;
using CallFuncArgs = std::vector<FuncArg>;
bool matchArgPattern(
    Value* v,
    const AtenFuncArgs& aten_func_args,
    const CallFuncArgs& call_func_args) {
```

and then

```
  return matchArgPattern(v, AtenFuncArgs{{"conv2d", 2}, {"linear", 2}}, CallFuncArgs{{"linear", 3}});
```

Or something liek that